realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/mongodb/sync/SyncedRealmTests.kt

```diff
@@ -655,6 +680,18 @@ class SyncedRealmTests {
                         assertEquals(expectedObjectIdList, it.columnObjectIdList)
                         assertEquals(expectedUUIDList, it.columnUUIDList)
                         assertEquals(expectedMixedList, it.columnMixedList)
+
+                        assertEquals(expectedObjectId, it.columnRealmDictionary["key"]!!.id)
```

Comment:
I know that it seems a bit boiler plate-ish to verify that the tests are exhaustive, but it is difficult to verify that you have covered all types from the above. Ex. `RealmDictionary<SyncDog>` seems to be left out.

Reply:
Have updated with an exhaustive switch-on-type.